### PR TITLE
Added multiple series for line and area graph

### PR DESCRIPTION
**Pull request by Ashvith**
Would you mind checking out the changes I've done for the `color` parameter in Vega?

### CODE REVIEW

**Comment by Ashvith** [comment]
I've not added tests for the same. Would be great if I could get some hints on the same.

File: .gitignore [hunk 1]
```diff
@@ -7,3 +7,4 @@
 /spec/reports/
 /tmp/
 *.lock
+*.gem
```
Comment (ankane) [inline .gitignore:10]: Please don't make changes to extra files.

File: .rubocop.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+AllCops:
+  DisabledByDefault: true
```
Comment (ankane) [inline .rubocop.yml:2]: Same comment as `.gitignore`.

File: lib/rover/data_frame.rb [hunk 3]
```diff
@@ -416,7 +416,7 @@ def plot(x = nil, y = nil, type: nil)
           raise "Cannot determine type. Use the type option."
         end
       end
-      data = self[[x, y]]
+      data = self[group.nil? ? [x, y] : [x, y, group]]
 
       case type
       when "line", "area"
```
Comment (ankane) [inline lib/rover/data_frame.rb:422]: Let's add it to all chart types. For charts that don't support it (I think just pie right now), we can raise an `ArgumentError`.

File: lib/rover/data_frame.rb [hunk 4]
```diff
@@ -436,9 +436,10 @@ def plot(x = nil, y = nil, type: nil)
           .mark(type: type, tooltip: true, interpolate: "cardinal", point: {size: 60})
           .encoding(
             x: {field: x, type: x_type, scale: scale},
-            y: {field: y, type: "quantitative"}
+            y: {field: y, type: "quantitative"},
+            color: {field: group}
           )
-          .config(axis: {labelFontSize: 12})
+          .config(axis: {title: nil, labelFontSize: 12})
```
Comment (ankane) [inline lib/rover/data_frame.rb:442]: I think we'll want to keep the titles.

**Review by ankane — COMMENTED** [review]
Hey @Ashvith, looks like a great start. Added some comments inline. For testing, we can follow the existing approach for now (`assert_plot_type` to make sure it doesn't error), and `assert_raises` for pie charts.

File: lib/rover/data_frame.rb (hunk 4 above)
Comment (Ashvith) [reply lib/rover/data_frame.rb:442]: Should I make it such that we can use variable parameters to decide whether to add title or not?

File: lib/rover/data_frame.rb (hunk 3 above)
Comment (Ashvith) [reply lib/rover/data_frame.rb:422]: But if we use `self[[x, y, group]]`, and don't pass the `group` argument, it throws an error.

**Comment by Ashvith** [comment]
Scatter plot isn't working properly. Maybe the problem is with the dataset I am trying out, so I probably have to find out another one that is compatible with the same.

**Comment by Ashvith** [comment]
> Hey @Ashvith, looks like a great start. Added some comments inline. For testing, we can follow the existing approach for now (`assert_plot_type` to make sure it doesn't error), and `assert_raises` for pie charts.

Can I insert a CSV file in the `test` directory?

File: lib/rover/data_frame.rb (hunk 3 above)
Comment (ankane) [reply lib/rover/data_frame.rb:422]: Yeah, that line looks good (comment was meant for `case` statement).

File: lib/rover/data_frame.rb [hunk 5]
```diff
@@ -436,7 +436,8 @@ def plot(x = nil, y = nil, type: nil)
           .mark(type: type, tooltip: true, interpolate: "cardinal", point: {size: 60})
           .encoding(
             x: {field: x, type: x_type, scale: scale},
-            y: {field: y, type: "quantitative"}
+            y: {field: y, type: "quantitative"},
+            color: {field: group}
```
Comment (ankane) [inline lib/rover/data_frame.rb:440]: Sorry forgot to mention last time: we should only add the `color` option if `group` isn't `nil`.

**Review by ankane — COMMENTED** [review]
Added a few more comments. I don't think the tests should need a CSV (see the other plot tests).

File: lib/rover/data_frame.rb (hunk 4 above)
Comment (ankane) [reply lib/rover/data_frame.rb]: Users should be able to call `config(axis: {title: nil})` on the returned Vega object, so I don't think an option is needed right now.

File: lib/rover/data_frame.rb (hunk 5 above)
Comment (Ashvith) [reply lib/rover/data_frame.rb:440]: I think that it still works the same way regardless of adding a ternary operator.

**Comment by Ashvith** [comment]
@ankane could you check the latest commit when you're free? I've added some changes as per your previous review.

**Comment by ankane** [comment]
Thanks @Ashvith, that looks better. I think the outstanding items are test cases and an `ArgumentError` for pie charts - https://github.com/ankane/rover/pull/23#pullrequestreview-866814682.

File: lib/rover/data_frame.rb [hunk 6]
```diff
@@ -430,13 +430,13 @@ def plot(x = nil, y = nil, type: nil)
           end
 
         scale = x_type == "temporal" ? {type: "utc"} : {}
-
```
Comment (ankane) [inline lib/rover/data_frame.rb:433]: Please make sure there aren't unnecessary changes to lines.

Comment (Ashvith) [reply lib/rover/data_frame.rb:433]: Must've been an accidental keyboard tap. Added the newline again.

**Comment by Ashvith** [comment]
The scatter plot works, but it shows an empty chart. It only works with [cars.csv](https://gist.github.com/noamross/e5d3e859aa0c794be10b) as of now.

Screenshot 2022-02-06 at 10-51-12 rover_demo - Jupyter Notebook

**Comment by Ashvith** [comment]
Hey @ankane, would you check out the latest commit? I am a little skeptical about the test for scatter plot.
